struts2-jquery-plugin not compatible with struts 6.0.0

Hello, today I updated struts from 2.5.30 to 6.0.0.

My app compiles and starts fine but when I try to access a page I get the following error:
```
o.a.struts2.components.ClosingUIBean:57 - Could not open template
freemarker.core.ParseException: Syntax error in template "template/jquery/autocompleter.ftl" in line 30, column 47:
Using ?html (legacy escaping) is not allowed when auto-escaping is on with a markup output format (HTML), to avoid double-escaping mistakes.
```

I tried setting the flag `struts.ui.escapeHtmlBody` to `false` but this did not solve the problem. Installing version 4.1.0-SNAPSHOT did not help either.
Do you know if it would somehow be possible to use the plugin with the newest struts? Maybe there is some other flag I overlooked?

When I add the property `<struts2.version>6.0.0</struts2.version>` to the pom of `struts2-jquery-integration-tests` and run `mvn verify` I get lots of similar errors, for example:
```
Caused by: freemarker.core.ParseException: Syntax error in template "template/jquery/div.ftl" in line 22, column 43:
Using ?html (legacy escaping) is not allowed when auto-escaping is on with a markup output format (HTML), to avoid double-escaping mistakes.
```

--- a/pom.xml
+++ b/pom.xml
@@ -3,7 +3,7 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>com.jgeppert.struts2.jquery</groupId>
 	<artifactId>struts2-jquery</artifactId>
-	<version>5.0.1-SNAPSHOT</version>
+	<version>5.0.0-SNAPSHOT</version>
 	<name>Struts 2 jQuery Plugin Parent Module</name>
 	<packaging>pom</packaging>
 	<url>https://github.com/struts-community-plugins/struts2-jquery</url>
@@ -84,7 +84,7 @@
 		<maven-war-plugin.version>3.3.2</maven-war-plugin.version>
 		<jetty.plugin.version>9.4.46.v20220331</jetty.plugin.version>
 		<!-- dependency versions -->
-		<struts2.version>2.5.26</struts2.version>
+		<struts2.version>6.0.0</struts2.version>
 		<tlib.version>2.3</tlib.version>
 		<minify.version>1.7.4</minify.version>
 		<beanutils.version>1.9.4</beanutils.version>

--- a/pom.xml
+++ b/pom.xml
@@ -1,5 +1,7 @@
 <?xml version="1.0"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>com.jgeppert.struts2.jquery</groupId>
 	<artifactId>struts2-jquery</artifactId>
@@ -63,10 +65,10 @@
 			<name>Struts 2 jQuery Plugin Staging Repository</name>
 			<url>https://oss.sonatype.org/service/local/staging/deploy/maven2</url>
 		</repository>
-		<snapshotRepository>			
+		<snapshotRepository>
 			<id>struts2-jquery.snapshots</id>
 			<name>Struts 2 jQuery Plugin Snapshots and Test Builds Repository</name>
-            <url>https://oss.sonatype.org/content/repositories/snapshots</url>			
+			<url>https://oss.sonatype.org/content/repositories/snapshots</url>
 			<uniqueVersion>false</uniqueVersion>
 		</snapshotRepository>
 	</distributionManagement>
@@ -95,7 +97,7 @@
 	<modules>
 		<module>struts2-jquery-plugin</module>
 		<module>struts2-jquery-grid-plugin</module>
-		<module>struts2-jquery-datatables-plugin</module>		
+		<module>struts2-jquery-datatables-plugin</module>
 		<module>struts2-jquery-richtext-plugin</module>
 		<module>struts2-jquery-chart-plugin</module>
 		<module>struts2-jquery-tree-plugin</module>
@@ -174,7 +176,7 @@
 			</plugin>
 		</plugins>
 		<defaultGoal>install</defaultGoal>
-		
+
 		<pluginManagement>
 			<plugins>
 				<plugin>
@@ -232,4 +234,15 @@
 		</dependency>
 
 	</dependencies>
+
+	<dependencyManagement>
+		<dependencies>
+			<dependency>
+				<groupId>org.apache.struts</groupId>
+				<artifactId>struts2-bom</artifactId>
+				<version>${struts2.version}</version>
+				<type>pom</type>
+			</dependency>
+		</dependencies>
+	</dependencyManagement>
 </project>

--- a/struts2-jquery-archetypes/pom.xml
+++ b/struts2-jquery-archetypes/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <groupId>com.jgeppert.struts2.jquery</groupId>
     <artifactId>struts2-jquery-archetypes</artifactId>

--- a/struts2-jquery-chart-plugin/pom.xml
+++ b/struts2-jquery-chart-plugin/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-chart-plugin</artifactId>
     <name>Struts 2 jQuery Chart Plugin</name>

--- a/struts2-jquery-chart-plugin/pom.xml
+++ b/struts2-jquery-chart-plugin/pom.xml
@@ -110,9 +110,9 @@
         </dependency>
 
         <dependency>
-            <groupId>org.apache.velocity</groupId>
-            <artifactId>velocity</artifactId>
-            <version>1.5</version>
+            <groupId>org.apache.struts</groupId>
+            <artifactId>struts2-velocity-plugin</artifactId>
+            <version>${struts2.version}</version>
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>

--- a/struts2-jquery-chart-plugin/src/main/java/com/jgeppert/struts2/jquery/chart/views/JqueryChartTagLibrary.java
+++ b/struts2-jquery-chart-plugin/src/main/java/com/jgeppert/struts2/jquery/chart/views/JqueryChartTagLibrary.java
@@ -42,7 +42,7 @@ public class JqueryChartTagLibrary implements TagLibraryDirectiveProvider, TagLi
     }
 
     @SuppressWarnings("rawtypes")
-    public List<Class> getDirectiveClasses() {
+    public List<Class<?>> getDirectiveClasses() {
         Class[] directives = new Class[]{ChartDirective.class,
                 ChartDataDirective.class};
         return Arrays.asList(directives);

--- a/struts2-jquery-datatables-plugin/pom.xml
+++ b/struts2-jquery-datatables-plugin/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-datatables-plugin</artifactId>
     <name>Struts 2 jQuery DataTables Plugin</name>

--- a/struts2-jquery-datatables-plugin/pom.xml
+++ b/struts2-jquery-datatables-plugin/pom.xml
@@ -110,9 +110,9 @@
         </dependency>
 
         <dependency>
-            <groupId>org.apache.velocity</groupId>
-            <artifactId>velocity</artifactId>
-            <version>1.5</version>
+            <groupId>org.apache.struts</groupId>
+            <artifactId>struts2-velocity-plugin</artifactId>
+            <version>${struts2.version}</version>
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>

--- a/struts2-jquery-datatables-plugin/src/main/java/com/jgeppert/struts2/jquery/datatables/views/JqueryDatatablesTagLibrary.java
+++ b/struts2-jquery-datatables-plugin/src/main/java/com/jgeppert/struts2/jquery/datatables/views/JqueryDatatablesTagLibrary.java
@@ -42,7 +42,7 @@ public class JqueryDatatablesTagLibrary implements TagLibraryDirectiveProvider, 
 
     @Override
     @SuppressWarnings("rawtypes")
-    public List<Class> getDirectiveClasses() {
+    public List<Class<?>> getDirectiveClasses() {
         Class[] directives = new Class[]{DatatablesDirective.class};
         return Arrays.asList(directives);
     }

--- a/struts2-jquery-grid-plugin/pom.xml
+++ b/struts2-jquery-grid-plugin/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-grid-plugin</artifactId>
     <name>Struts 2 jQuery Grid Plugin</name>

--- a/struts2-jquery-grid-plugin/pom.xml
+++ b/struts2-jquery-grid-plugin/pom.xml
@@ -118,9 +118,9 @@
         </dependency>
 
         <dependency>
-            <groupId>org.apache.velocity</groupId>
-            <artifactId>velocity</artifactId>
-            <version>1.5</version>
+            <groupId>org.apache.struts</groupId>
+            <artifactId>struts2-velocity-plugin</artifactId>
+            <version>${struts2.version}</version>
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>

--- a/struts2-jquery-grid-plugin/src/main/java/com/jgeppert/struts2/jquery/grid/views/JqueryGridTagLibrary.java
+++ b/struts2-jquery-grid-plugin/src/main/java/com/jgeppert/struts2/jquery/grid/views/JqueryGridTagLibrary.java
@@ -41,7 +41,7 @@ public class JqueryGridTagLibrary implements TagLibraryDirectiveProvider, TagLib
     }
 
     @SuppressWarnings("rawtypes")
-    public List<Class> getDirectiveClasses() {
+    public List<Class<?>> getDirectiveClasses() {
         Class[] directives = new Class[]{
                 GridDirective.class, GridColumnDirective.class,
         };

--- a/struts2-jquery-grid-showcase/pom.xml
+++ b/struts2-jquery-grid-showcase/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-grid-showcase</artifactId>
     <name>Struts 2 jQuery Grid Showcase</name>

--- a/struts2-jquery-integration-tests/pom.xml
+++ b/struts2-jquery-integration-tests/pom.xml
@@ -5,7 +5,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-integration-tests</artifactId>
     <name>Struts 2 jQuery Integration Tests</name>

--- a/struts2-jquery-mobile-plugin/pom.xml
+++ b/struts2-jquery-mobile-plugin/pom.xml
@@ -125,9 +125,9 @@
         </dependency>
 
         <dependency>
-            <groupId>org.apache.velocity</groupId>
-            <artifactId>velocity</artifactId>
-            <version>1.5</version>
+            <groupId>org.apache.struts</groupId>
+            <artifactId>struts2-velocity-plugin</artifactId>
+            <version>${struts2.version}</version>
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>

--- a/struts2-jquery-mobile-plugin/pom.xml
+++ b/struts2-jquery-mobile-plugin/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-mobile-plugin</artifactId>
     <name>Struts 2 jQuery Mobile Plugin</name>

--- a/struts2-jquery-mobile-plugin/src/main/java/com/jgeppert/struts2/jquery/mobile/views/JqueryMobileTagLibrary.java
+++ b/struts2-jquery-mobile-plugin/src/main/java/com/jgeppert/struts2/jquery/mobile/views/JqueryMobileTagLibrary.java
@@ -42,7 +42,7 @@ public class JqueryMobileTagLibrary implements TagLibraryDirectiveProvider, TagL
     }
 
     @SuppressWarnings("rawtypes")
-    public List<Class> getDirectiveClasses() {
+    public List<Class<?>> getDirectiveClasses() {
         Class[] directives = new Class[]{HeadDirective.class,
                 CheckboxListDirective.class, CheckboxDirective.class,
                 Radio.class, DivDirective.class, ListDirective.class,

--- a/struts2-jquery-mobile-showcase/pom.xml
+++ b/struts2-jquery-mobile-showcase/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-mobile-showcase</artifactId>
     <name>Struts 2 jQuery Mobile Showcase</name>

--- a/struts2-jquery-plugin/pom.xml
+++ b/struts2-jquery-plugin/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-plugin</artifactId>
     <name>Struts 2 jQuery Plugin</name>

--- a/struts2-jquery-plugin/pom.xml
+++ b/struts2-jquery-plugin/pom.xml
@@ -245,9 +245,9 @@
             <scope>compile</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.velocity</groupId>
-            <artifactId>velocity</artifactId>
-            <version>1.5</version>
+            <groupId>org.apache.struts</groupId>
+            <artifactId>struts2-velocity-plugin</artifactId>
+            <version>${struts2.version}</version>
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>

--- a/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/components/AbstractFormListElement.java
+++ b/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/components/AbstractFormListElement.java
@@ -105,13 +105,13 @@ public abstract class AbstractFormListElement extends AbstractFormElement {
             }
 
             if (listKey != null) {
-                addParameter(PARAM_LIST_KEY, stripExpressionIfAltSyntax(listKey));
+                addParameter(PARAM_LIST_KEY, stripExpression(listKey));
             } else if (value instanceof Map) {
                 addParameter(PARAM_LIST_KEY, "key");
             }
 
             if (listValue != null) {
-                addParameter(PARAM_LIST_VALUE, stripExpressionIfAltSyntax(listValue));
+                addParameter(PARAM_LIST_VALUE, stripExpression(listValue));
             } else if (value instanceof Map) {
                 addParameter(PARAM_LIST_VALUE, "value");
             }

--- a/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/components/Accordion.java
+++ b/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/components/Accordion.java
@@ -211,14 +211,14 @@ public class Accordion extends AbstractTopicsBean {
         }
 
         if (listKey != null) {
-            listKey = stripExpressionIfAltSyntax(listKey);
+            listKey = stripExpression(listKey);
             addParameter(PARAM_LIST_KEY, listKey);
         } else if (value instanceof Map) {
             addParameter(PARAM_LIST_KEY, "key");
         }
 
         if (listValue != null) {
-            listValue = stripExpressionIfAltSyntax(listValue);
+            listValue = stripExpression(listValue);
             addParameter(PARAM_LIST_VALUE, listValue);
         } else if (value instanceof Map) {
             addParameter(PARAM_LIST_VALUE, "value");

--- a/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/components/Autocompleter.java
+++ b/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/components/Autocompleter.java
@@ -159,7 +159,7 @@ public class Autocompleter extends AbstractFormListElement {
         addParameterIfPresent(PARAM_AUTO_FOCUS, this.autoFocus, Boolean.class);
 
         if (listLabel != null) {
-            addParameter(PARAM_LIST_LABEL, stripExpressionIfAltSyntax(listLabel));
+            addParameter(PARAM_LIST_LABEL, stripExpression(listLabel));
         }
 
         addGeneratedIdParam(ID_PREFIX_AUTOCOMPLETER);

--- a/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/components/Menu.java
+++ b/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/components/Menu.java
@@ -175,14 +175,14 @@ public class Menu extends AbstractTopicsBean {
         }
 
         if (listKey != null) {
-            listKey = stripExpressionIfAltSyntax(listKey);
+            listKey = stripExpression(listKey);
             addParameter(PARAM_LIST_KEY, listKey);
         } else if (value instanceof Map) {
             addParameter(PARAM_LIST_KEY, "key");
         }
 
         if (listValue != null) {
-            listValue = stripExpressionIfAltSyntax(listValue);
+            listValue = stripExpression(listValue);
             addParameter(PARAM_LIST_VALUE, listValue);
         } else if (value instanceof Map) {
             addParameter(PARAM_LIST_VALUE, "value");

--- a/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/views/JqueryTagLibrary.java
+++ b/struts2-jquery-plugin/src/main/java/com/jgeppert/struts2/jquery/views/JqueryTagLibrary.java
@@ -41,7 +41,7 @@ public class JqueryTagLibrary implements TagLibraryDirectiveProvider, TagLibrary
     }
 
     @SuppressWarnings("rawtypes")
-    public List<Class> getDirectiveClasses() {
+    public List<Class<?>> getDirectiveClasses() {
         Class[] directives =
                 new Class[]{
                         DatePickerDirective.class,

--- a/struts2-jquery-richtext-plugin/pom.xml
+++ b/struts2-jquery-richtext-plugin/pom.xml
@@ -108,9 +108,9 @@
             <scope>compile</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.velocity</groupId>
-            <artifactId>velocity</artifactId>
-            <version>1.5</version>
+            <groupId>org.apache.struts</groupId>
+            <artifactId>struts2-velocity-plugin</artifactId>
+            <version>${struts2.version}</version>
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>

--- a/struts2-jquery-richtext-plugin/pom.xml
+++ b/struts2-jquery-richtext-plugin/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-richtext-plugin</artifactId>
     <name>Struts 2 jQuery Richtext Plugin</name>

--- a/struts2-jquery-richtext-plugin/src/main/java/com/jgeppert/struts2/jquery/richtext/views/JqueryRichtextTagLibrary.java
+++ b/struts2-jquery-richtext-plugin/src/main/java/com/jgeppert/struts2/jquery/richtext/views/JqueryRichtextTagLibrary.java
@@ -41,7 +41,7 @@ public class JqueryRichtextTagLibrary implements TagLibraryDirectiveProvider, Ta
     }
 
     @SuppressWarnings("rawtypes")
-    public List<Class> getDirectiveClasses() {
+    public List<Class<?>> getDirectiveClasses() {
         Class[] directives = new Class[]{
                 CkeditorDirective.class, TinymceDirective.class
         };

--- a/struts2-jquery-showcase/pom.xml
+++ b/struts2-jquery-showcase/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-showcase</artifactId>
     <name>Struts 2 jQuery Showcase</name>

--- a/struts2-jquery-tree-plugin/pom.xml
+++ b/struts2-jquery-tree-plugin/pom.xml
@@ -108,9 +108,9 @@
             <scope>compile</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.velocity</groupId>
-            <artifactId>velocity</artifactId>
-            <version>1.5</version>
+            <groupId>org.apache.struts</groupId>
+            <artifactId>struts2-velocity-plugin</artifactId>
+            <version>${struts2.version}</version>
             <scope>compile</scope>
             <optional>true</optional>
         </dependency>

--- a/struts2-jquery-tree-plugin/pom.xml
+++ b/struts2-jquery-tree-plugin/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <groupId>com.jgeppert.struts2.jquery</groupId>
         <artifactId>struts2-jquery</artifactId>
-        <version>4.1.0-SNAPSHOT</version>
+        <version>5.0.0-SNAPSHOT</version>
     </parent>
     <artifactId>struts2-jquery-tree-plugin</artifactId>
     <name>Struts 2 jQuery Tree Plugin</name>

--- a/struts2-jquery-tree-plugin/src/main/java/com/jgeppert/struts2/jquery/tree/views/JqueryTreeTagLibrary.java
+++ b/struts2-jquery-tree-plugin/src/main/java/com/jgeppert/struts2/jquery/tree/views/JqueryTreeTagLibrary.java
@@ -42,7 +42,7 @@ public class JqueryTreeTagLibrary implements TagLibraryDirectiveProvider, TagLib
     }
 
     @SuppressWarnings("rawtypes")
-    public List<Class> getDirectiveClasses() {
+    public List<Class<?>> getDirectiveClasses() {
         Class[] directives = new Class[]{TreeDirective.class,
                 TreeItemDirective.class};
         return Arrays.asList(directives);

